language/move-vm/runtime/src/debug.rs

```
@@ -0,0 +1,194 @@
+                    .clone();
+                println!(
+                    "Breakpoint {} hit with instruction {}",
+                    bp_match, instr_string
+                );
+            }
+            println!(
+                "function >> {}\ninstruction >> {:?}\nprogram counter >> {}",
+                function_string, instr, pc
+            );
+            loop {
+                print!("> ");
+                std::io::stdout().flush().unwrap();
+                let mut input = String::new();
+                match io::stdin().read_line(&mut input) {
+                    Ok(_) => match input.parse::<DebugCommand>() {
+                        Err(err) => println!("{}", err),
+                        Ok(command) => match command {
+                            DebugCommand::Step => {
+                                self.should_take_input = true;
+                                break;
+                            }
+                            DebugCommand::Continue => {
+                                self.should_take_input = false;
+                                break;
+                            }
+                            DebugCommand::Breakpoint(breakpoint) => {
+                                self.breakpoints.insert(breakpoint.to_string());
+                            }
+                            DebugCommand::DeleteBreakpoint(breakpoint) => {
+                                self.breakpoints.remove(&breakpoint);
+                            }
+                            DebugCommand::PrintBreakpoints => self
+                                .breakpoints
+                                .iter()
+                                .enumerate()
+                                .for_each(|(i, bp)| println!("[{}] {}", i, bp)),
+                            DebugCommand::PrintStack => {
+                                let mut s = String::new();
+                                let _ = interp.debug_print_stack_trace(&mut s, resolver);
```

So here we're calling a function with side effects and silently discarding its result. A little bit disturbed by this but I understand you can't simply propagate the error up. Would it be better we `unwrap` here? I assume it's fine for it to crash in case something goes wrong in `debug_print_stack_trace` as it only happens in a non-prod build.

---

Good spot, I'll add that `unwrap` in!